From 2092e109c77e13358716ee9a307316f804ffd99e Mon Sep 17 00:00:00 2001
From: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
Date: Tue, 24 Jul 2018 21:44:38 -0700
Subject: [PATCH 1593/1633] ASoC: SOF: modify load_firmware() to indicate first
 boot

Modify the signature for load_firmware to add a flag indicating
if this is the first time the DSP is being booted up.
Based on the platform specific implementation, this flag can be
used to decide whether the firmware can be booted from memory
in self-retention or to be downloaded again.

Note that this patch does not change the implementation of any
platform specific load_firmware() method. It merely adds a flag
that can be used to modify the implementation later if needed.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
---
 sound/soc/sof/core.c             | 2 +-
 sound/soc/sof/intel/hda-loader.c | 3 ++-
 sound/soc/sof/intel/hda.h        | 3 ++-
 sound/soc/sof/loader.c           | 6 +++---
 sound/soc/sof/sof-priv.h         | 6 +++---
 5 files changed, 11 insertions(+), 9 deletions(-)

diff --git a/sound/soc/sof/core.c b/sound/soc/sof/core.c
index 6cdd3ea..924220c5 100644
--- a/sound/soc/sof/core.c
+++ b/sound/soc/sof/core.c
@@ -292,7 +292,7 @@ static int sof_probe(struct platform_device *pdev)
 	}
 
 	/* load the firmware */
-	ret = snd_sof_load_firmware(sdev, plat_data->fw);
+	ret = snd_sof_load_firmware(sdev, plat_data->fw, true);
 	if (ret < 0) {
 		dev_err(sdev->dev, "error: failed to load DSP firmware %d\n",
 			ret);
diff --git a/sound/soc/sof/intel/hda-loader.c b/sound/soc/sof/intel/hda-loader.c
index 372ccfa..d5476bb 100644
--- a/sound/soc/sof/intel/hda-loader.c
+++ b/sound/soc/sof/intel/hda-loader.c
@@ -287,7 +287,8 @@ static int cl_copy_fw(struct snd_sof_dev *sdev, int tag)
 	return status;
 }
 
-int hda_dsp_cl_load_fw(struct snd_sof_dev *sdev, const struct firmware *fw)
+int hda_dsp_cl_load_fw(struct snd_sof_dev *sdev, const struct firmware *fw,
+		       bool first_boot)
 {
 	struct snd_sof_pdata *plat_data = dev_get_platdata(sdev->dev);
 	int ret;
diff --git a/sound/soc/sof/intel/hda.h b/sound/soc/sof/intel/hda.h
index 8d353ac..2f3bc45 100644
--- a/sound/soc/sof/intel/hda.h
+++ b/sound/soc/sof/intel/hda.h
@@ -483,7 +483,8 @@ int hda_dsp_ipc_cmd_done(struct snd_sof_dev *sdev, int dir);
 /*
  * DSP Code loader.
  */
-int hda_dsp_cl_load_fw(struct snd_sof_dev *sdev, const struct firmware *fw);
+int hda_dsp_cl_load_fw(struct snd_sof_dev *sdev, const struct firmware *fw,
+		       bool first_boot);
 int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev);
 
 /*
diff --git a/sound/soc/sof/loader.c b/sound/soc/sof/loader.c
index 4bc7d7a..210a8d3 100644
--- a/sound/soc/sof/loader.c
+++ b/sound/soc/sof/loader.c
@@ -208,7 +208,7 @@ static int load_modules(struct snd_sof_dev *sdev, const struct firmware *fw)
 }
 
 int snd_sof_load_firmware_memcpy(struct snd_sof_dev *sdev,
-				 const struct firmware *fw)
+				 const struct firmware *fw, bool first_boot)
 {
 	int ret;
 
@@ -238,12 +238,12 @@ int snd_sof_load_firmware_memcpy(struct snd_sof_dev *sdev,
 EXPORT_SYMBOL(snd_sof_load_firmware_memcpy);
 
 int snd_sof_load_firmware(struct snd_sof_dev *sdev,
-			  const struct firmware *fw)
+			  const struct firmware *fw, bool first_boot)
 {
 	dev_dbg(sdev->dev, "loading firmware\n");
 
 	if (sdev->ops->load_firmware)
-		return sdev->ops->load_firmware(sdev, fw);
+		return sdev->ops->load_firmware(sdev, fw, first_boot);
 	return 0;
 }
 EXPORT_SYMBOL(snd_sof_load_firmware);
diff --git a/sound/soc/sof/sof-priv.h b/sound/soc/sof/sof-priv.h
index 716e590..c382754 100644
--- a/sound/soc/sof/sof-priv.h
+++ b/sound/soc/sof/sof-priv.h
@@ -140,7 +140,7 @@ struct snd_sof_dsp_ops {
 
 	/* FW loading */
 	int (*load_firmware)(struct snd_sof_dev *sof_dev,
-			     const struct firmware *fw);
+			     const struct firmware *fw, bool first_boot);
 	int (*load_module)(struct snd_sof_dev *sof_dev,
 			   struct snd_sof_mod_hdr *hdr);
 	int (*fw_ready)(struct snd_sof_dev *sdev, u32 msg_id);
@@ -391,9 +391,9 @@ int snd_sof_create_page_table(struct snd_sof_dev *sdev,
  * Firmware loading.
  */
 int snd_sof_load_firmware(struct snd_sof_dev *sdev,
-			  const struct firmware *fw);
+			  const struct firmware *fw, bool first_boot);
 int snd_sof_load_firmware_memcpy(struct snd_sof_dev *sdev,
-				 const struct firmware *fw);
+				 const struct firmware *fw, bool first_boot);
 int snd_sof_run_firmware(struct snd_sof_dev *sdev);
 int snd_sof_parse_module_memcpy(struct snd_sof_dev *sdev,
 				struct snd_sof_mod_hdr *module);
-- 
2.7.4

